### PR TITLE
OLS-293: Sequence diagram with basic flow

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Sequence diagram with basic flow

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [x] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-293](https://issues.redhat.com//browse/OLS-293)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Comment by onmete** [comment]
Cool, we just need to remember to update it when some flow changes.

/lgtm

File: docs/sequence_diagram.uml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+//
```
Comment (vishnuchalla) [inline docs/sequence_diagram.uml:1]: I like this idea. Thanks for setting this up.
/lgtm

File: README.md [hunk 2]
```diff
@@ -163,6 +163,25 @@ helm delete ols-release --namespace openshift-lightspeed
...
 
 
+## Project structure
+
+1. REST API handlers
+1. Configuration loader
+1. LLM loader
+1. Doc retriever from vector storage
+1. Question validator
+1. Docs summarizer
```
Comment (bparees) [inline README.md:173]: we need to come up with a better name for this component.  Right now it's really more of a "generic question answerer" (it uses RAG content when available, but if not it just sends the user question on to the LLM as is)

File: docs/sequence_diagram.uml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+alt "Invalid question case"
+question_validator -> ols: Invalid question
+ols -> user: Your question is not valid
+end
+
+alt "Valid question case"
+question_validator -> ols: Valid question
+ols -> document_summarizer: Summarize document
+document_summarizer -> llm: Query
+llm -> document_summarizer: Answer
```
Comment (bparees) [inline docs/sequence_diagram.uml:49]: How about also showing the RAG retrieval piece of the flow?  You mentioned the docs retriever as a component but then didn't include it in the flow diagram

Note that rag retrieval is currently "optional" (if there's no RAG content configured, we skip the retrieval step but still send the user query to the llm)

Comment (tisnik) [reply docs/sequence_diagram.uml:49]: yup, I'll add it

Comment (tisnik) [reply docs/sequence_diagram.uml:52]: updated, but not sure about naming @bparees

Comment (bparees) [reply docs/sequence_diagram.uml:52]: good enough for now, thanks

**Pull request merged**